ckanext/canada/logic.py

```diff
@@ -185,3 +194,88 @@ def datastore_create_temp_user_table(context):
             username=literal_string(username),
             sysadmin='TRUE' if is_sysadmin(username) else 'FALSE'))
 
+
+@chained_action
+def canada_resource_view_show(up_func, context, data_dict):
+    """
+    Return the metadata of a resource_view.
+
+    :param id: the id of the resource_view
+    :type id: string
+
+    :rtype: dictionary
+
+    Canada Fork: extends the resource_view_show action method
+    to prevent showing datatable_views for invalid and inactive Resources
+
+    We need this method to 404 the resource_view page (e.g. viewing in full screen)
+
+    If a user is logged in, we want them to be able to see and edit the datatables_view
+    views still. We will just add a key to the view dict to be used within templates for visuals.
+    """
+    view_dict = up_func(context, data_dict)
+    if view_dict.get('view_type') == 'datatables_view':
+        # at this point, the core function has been called, calling resource_show
+        # and adding the Resource Object into the context
+        res_dict = context.get('resource').__dict__
+        res_extras = res_dict.get('extras', {})
+        site_user = get_action('get_site_user')({'ignore_auth': True}, {})['name']
+
+        if not res_extras.get('datastore_active', False) or \
+        res_extras.get('validation_status') != 'success':
+            # if the Resource is not active in the DataStore or has not
+            # passed validation, we do not want to show its datatables_view views
+
+            if not g.user:
+                # only raise if a user is not logged in
+                raise ObjectNotFound
+
+            if g.user != site_user:
+                # only add key/value if not system process
+                view_dict['canada_disabled_view'] = True
+
+    return view_dict
+
+
+@chained_action
+def canada_resource_view_list(up_func, context, data_dict):
+    """
+    Return the list of resource views for a particular resource.
+
+    :param id: the id of the resource
+    :type id: string
+
+    :rtype: list of dictionaries.
+
+    Canada Fork: extends the resource_view_list action method
+    to prevent showing datatable_views for invalid and inactive Resources
+
+    This will delete any datatables_view from the list of views.
+
+    If a user is logged in, we want them to be able to see and edit the datatables_view
+    views still. We will just add a key to the view dict to be used within templates for visuals.
+    """
+    view_list = up_func(context, data_dict)
+    # at this point, the core function has been called, calling resource_show
+    # and adding the Resource Object into the context
+    res_dict = context.get('resource').__dict__
+    res_extras = res_dict.get('extras', {})
```

Comment:
same here, there's rarely a good reason to access `__dict__`

Reply:
yup, not sure why I did that. I can just access `context.get('resource').extras` right??

Reply:
oops sorry did not see ur above suggestion. sweeeeet